components/telemetry-operator/apis/telemetry/v1alpha1/logpipeline_types.go

```diff
@@ -140,28 +140,32 @@ type FileMount struct {
 
 // VariableReference references a Kubernetes secret that should be provided as environment variable to Fluent Bit
 type VariableReference struct {
-	Name      string        `json:"name,omitempty"`
-	ValueFrom ValueFromType `json:"valueFrom,omitempty"`
+	Name      string          `json:"name,omitempty"`
+	ValueFrom ValueFromSource `json:"valueFrom,omitempty"`
 }
 
 type ValueType struct {
-	Value     string        `json:"value,omitempty"`
-	ValueFrom ValueFromType `json:"valueFrom,omitempty"`
+	Value     string           `json:"value,omitempty"`
+	ValueFrom *ValueFromSource `json:"valueFrom,omitempty"`
 }
 
 func (v *ValueType) IsDefined() bool {
-	return v.Value != "" || v.ValueFrom.IsSecretRef()
+	if v.Value != "" {
+		return true
+	}
+
+	return v.ValueFrom != nil && v.ValueFrom.IsSecretRef()
 }
 
-func (v *ValueFromType) IsSecretRef() bool {
-	return v.SecretKey.Name != "" && v.SecretKey.Key != ""
+type ValueFromSource struct {
+	SecretRef *SecretRef `json:"secretKeyRef,omitempty"`
 }
 
-type ValueFromType struct {
-	SecretKey SecretKeyRef `json:"secretKeyRef,omitempty"`
+func (v *ValueFromSource) IsSecretRef() bool {
```

Comment:
Maybe this should be on the ValueType? Why ask the non-nil SecretRef if it "IsSecretRef"? Or rename it to something else if it's just for validation?

Reply:
After a discussion, we decided to keep it like this so far